tests/integration/site-name/site-name.test.js

```diff
@@ -0,0 +1,205 @@
+import { JSDOM } from 'jsdom'
+import { getByRole, getByText } from '@testing-library/dom'
+import { createServer } from '~/src/server/index.js'
+import { statusCodes } from '~/src/server/common/constants/status-codes.js'
+import {
+  getExemptionCache,
+  updateExemptionSiteDetails,
+  setExemptionCache
+} from '~/src/server/common/helpers/session-cache/utils.js'
+import { validateErrors } from '../utils/utils.test.js'
+
+jest.mock('~/src/server/common/helpers/session-cache/utils.js')
+
+describe('Site name page', () => {
+  let server
+
+  const mockExemption = {
+    id: 'test-exemption-123',
+    projectName: 'Test Project'
+  }
+
+  jest.mocked(getExemptionCache).mockReturnValue(mockExemption)
+  jest.mocked(updateExemptionSiteDetails).mockReturnValue({})
+  jest.mocked(setExemptionCache).mockReturnValue({})
+
+  beforeAll(async () => {
+    server = await createServer()
+    await server.initialize()
+  })
+
+  afterAll(async () => {
+    await server.stop()
+  })
+
+  test('should display the site name page with correct content', async () => {
+    const { result, statusCode } = await server.inject({
+      method: 'GET',
+      url: '/exemption/site-name'
+    })
+
+    expect(statusCode).toBe(statusCodes.ok)
+
+    const { document } = new JSDOM(result).window
+
+    expect(
+      getByRole(document, 'heading', {
+        level: 1,
+        name: 'Site name'
+      })
+    ).toBeInTheDocument()
+    expect(getByText(document, mockExemption.projectName)).toBeInTheDocument()
+    expect(getByText(document, 'Site 1')).toBeInTheDocument()
+
+    const siteNameInput = document.querySelector('input[name="siteName"]')
+    expect(siteNameInput).toBeInTheDocument()
+    expect(siteNameInput).toHaveAttribute('type', 'text')
+
+    expect(
+      getByRole(document, 'button', { name: 'Continue' })
+    ).toBeInTheDocument()
+    expect(getByRole(document, 'link', { name: 'Cancel' })).toBeInTheDocument()
+    expect(getByRole(document, 'link', { name: 'Back' })).toBeInTheDocument()
+
+    expect(setExemptionCache).not.toHaveBeenCalled()
+  })
+
+  test('should pre-populate input when siteName value exists in cache', async () => {
+    jest.mocked(getExemptionCache).mockReturnValue({
+      ...mockExemption,
+      siteDetails: { ...mockExemption.siteDetails, siteName: 'Test Site' }
+    })
+
+    const { result, statusCode } = await server.inject({
+      method: 'GET',
+      url: '/exemption/site-name'
+    })
+
+    expect(statusCode).toBe(statusCodes.ok)
+
+    const { document } = new JSDOM(result).window
+
+    const siteNameInput = document.querySelector('input[name="siteName"]')
```

Comment:
`getByLabelText('Site name')` will return the input and also do a a11y check for free, that the label is associated with the input